### DISCUSSION TITLE
Deploy docker images to App Service Plan

### DISCUSSION THREAD

**Discussion by Digma** [Azure/azure-dev#2646] [Q&A]
Hi all,

We have some python apps (web apps and Azure function) running on an App Service plan. By default, the deployment of python pacakges to app service plan is done using the .zip approach. In our case, we would like to use docker container instead 

App Service plan support Docker Containers. However, azd does not allow (through `azure deploy`) to deploy Docker containers to app service (only to container apps).

Is there any workaround that could be used to keep the simple `azd up / azd deploy` flow working but use docker containers with the app service plan?

**Note**: There is an issue open about adding the functionality (https://github.com/Azure/azure-dev/issues/1045). The question would be about workaround (e.g. using hooks) that could allow without waiting for the feature described in #1045.

**Comment by vhvb1989** [comment]
You need the bicep definition for the `WebApps` with `app,linux,container` and the runtime `DOCKER`.
Then, you need to either create a container-registry (where your docker-img will be pushed/hosted) or define an existing registry.

**Comment by vhvb1989** [comment]
@Digma, I just tried it and `azd` can't currently handle the requested scenario.

When `azd` is configured to use `host=containerapp`, it will expect to find `azure container app` service, and will fail after finding `Azure Web Apps`

![image](https://github.com/Azure/azure-dev/assets/24213737/9e404e3d-c469-428d-af21-b51d7ff59143)


It is required a new feature for azd to support the scenario. That would be to support a new `host` like `appservice-container` or adding an extra field-config for host `appservice` to switch from the current supported mode (runtime-lang) to `docker` mode.

**Reply by Digma** [reply]
Thanks @vhvb1989 for the detailed answer. 

Indeed the scenario is not implemented and for now the best work-arounds are probably to push the docker images to a container beforehand and then provision the apps directly with the link to the container (`azd provision`) instead of relying on  `azd deploy`. 

Does the community think it is a valuable feature having the `host=appservice-container` option in azd? For us at least, standarizing our apps/apis/... as docker containers (instead of the .zip) is easier in general.
